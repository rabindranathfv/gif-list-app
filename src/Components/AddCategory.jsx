import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = (props) => {

    const { setCategory } = props;
    const [inputValue, setInputValue] = useState('');

    const addHandler = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 2) {
            setCategory(cat => [ inputValue]);
            console.log('AQUI');
            setInputValue('');
            console.log('AQUI 4');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            { inputValue && <p>{ inputValue }</p> }
            <input type="text" value={inputValue} onChange={addHandler} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}