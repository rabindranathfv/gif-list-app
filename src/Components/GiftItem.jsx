import React from 'react'
import PropTypes from 'prop-types';

export const GiftItem = ({ gift }) => {

    return (
        <div className="animate__animated animate__bounce grid-item" key={gift.id} >
            <p > {gift.title}</p>
            <img className='grid-image' src={gift.url} alt={gift.title} />
        </div>
    )
}

GiftItem.propTypes = {
    gift: PropTypes.shape({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired
    })
}