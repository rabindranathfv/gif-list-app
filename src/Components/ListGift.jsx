import React, { useState } from 'react';

export const ListGift = (props) => {

    const { gifts } = props;
    const [categories, setCategories] = useState(gifts);
    return (
        <div>
            <ul>
            {
                categories.map( cat => {
                    return <li key={cat.id}> {cat.name} </li>
                })
            }
            </ul>
        </div>
    )
}
