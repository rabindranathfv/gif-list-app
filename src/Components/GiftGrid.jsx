import React, { useEffect, useState } from 'react'
import PropTypes, { string } from 'prop-types';

import { GiftItem } from './GiftItem';

const API_URL = `https://api.giphy.com/v1/gifs/search`;
const API_KEY = `6ijpJmhvX9OshRF2lAHuBYHojzB9bZjX`;

export const GiftGrid = props => {

    const { category, searchCategory } = props;

    const [gifts, setGifts] = useState([]);

    const getGifts = async() => {
        const urlQuery = `${API_URL}?q=${searchCategory}&limit=10&api_key=${API_KEY}`
        const resp = await fetch( urlQuery);
        const { data } = await resp.json();
        
        const giftItem = data.map( gif => {
            return { 
                id: gif.id,
                title: gif.title,
                url:  gif.images?.downsized_medium.url
            }
        })
        setGifts( giftItem);
    }

    useEffect(() => {
        getGifts();
    }, []);

    return (
        <div className='card-grid'>
            <ul>
            {
                gifts.map( (gif) => {
                    return <GiftItem gift={gif} /> 
                })
            }
            </ul>
        </div>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.shape({
        id: string,
        title: string,
        url: string
      }).isRequired,
}
