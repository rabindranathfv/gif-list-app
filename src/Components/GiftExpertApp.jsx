import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = () => {

    const [category, setCategory] = useState('');

    console.log('CATEGORY IN MAIN COMPONENT', category);

    return (
        <div>
            <h2> GifExpertAPP </h2>
            <AddCategory setCategory={setCategory} />
            <hr />
            <GiftGrid category={category}  />
        </div>
    )
}

GiftExpertApp.defaultProps = {
    category: 'dragon ball',
}

export default GiftExpertApp
