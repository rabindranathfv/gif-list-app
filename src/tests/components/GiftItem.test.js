import { shallow } from 'enzyme';
import React from 'react';
import { GiftItem } from '../../Components/GiftItem';

describe('GiftItem', () => {
    const giftMock = {      
        id: 'GRSnxyhJnPsaQy9YLn',
        title: 'Dragon Ball GIF by TOEI Animation UK',
        url: 'https://media4.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=59b5fe65oh3yejavpf4jn4afobujki54npw5rtvcp6y2i2s6&rid=giphy.gif&ct=g'
    };
    let wrapper = shallow( <GiftItem gift={giftMock} />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <GiftItem gift={giftMock} />);
    })

    test('should render GiftItem', () => {
        expect( wrapper).toMatchSnapshot();
    })
    
    test('should render all gift info', () => {
        const titleGift = wrapper.find('p');
        const imgGift = wrapper.find('img');

        expect(titleGift.text().trim()).toBe(giftMock.title);
        expect(imgGift.prop('alt')).toBe(giftMock.title);
        expect(imgGift.prop('src')).toBe(giftMock.url);
    })
    
})
