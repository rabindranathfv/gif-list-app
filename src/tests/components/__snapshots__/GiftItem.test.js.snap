// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItem should render GiftItem 1`] = `
<div
  className="animate__animated animate__bounce grid-item"
  key="GRSnxyhJnPsaQy9YLn"
>
  <p>
     
    Dragon Ball GIF by TOEI Animation UK
  </p>
  <img
    alt="Dragon Ball GIF by TOEI Animation UK"
    className="grid-image"
    src="https://media4.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=59b5fe65oh3yejavpf4jn4afobujki54npw5rtvcp6y2i2s6&rid=giphy.gif&ct=g"
  />
</div>
`;
